src/clients/apiClient.ts

```diff
@@ -125,14 +131,13 @@ export async function createNewUser(newUser: NewUser): Promise<Response> {
   }
 }
 
-export interface SpeciesSearch {
-  tailType: number;
-  size: number;
-  colour: string;
-}
-
-export async function fetchSpeciesQuery(speciesSearch: SpeciesSearch): Promise<Response> {
-  const response = await fetch(`${backendUrl}/species?TailType=${speciesSearch.tailType}&Size=${speciesSearch.size}&Colour=${speciesSearch.colour}`);
+export async function fetchSpeciesQuery(speciesSearch: SpeciesSearch): Promise<WhaleSpecies[]> {
+  let response;
+  if (speciesSearch.tailType == null && speciesSearch.colour == null && speciesSearch.colour == null) {
+    response = await fetch(`${backendUrl}/species?TailType=&Size=&Colour=`);
+  } else {
+    response = await fetch(`${backendUrl}/species?TailType=${speciesSearch.tailType}&Size=${speciesSearch.size}&Colour=${speciesSearch.colour}`);
+  }
```

Comment:
Instead of handling it like this could we set the values of the variables individually and pass them into the URL?
```suggestion
  const tailType = speciesSearch.tailType == null ? "" : "speciesSearch.tailType";
  const colour = speciesSearch.colour == null ? "" : "speciesSearch.colour";
  const size = speciesSearch.size == null ? "" : "speciesSearch.size";
  const response = await fetch(`${backendUrl}/species?TailType=${tailType}&Size=${size}&Colour=${colour}`);
  }
```